Cython/Compiler/Main.py

```diff
@@ -91,6 +91,8 @@ def __init__(self, include_directories, compiler_directives, cpp=False,
         if language_level is not None:
             self.set_language_level(language_level)
 
+        self.legacy_implicit_noexcept = getattr(options, 'legacy_implicit_noexcept', False)
```

Comment:
Is the `getattr()` only needed for the `None` case? If so, I'd change it to be more explicit:
```suggestion
        self.legacy_implicit_noexcept = options.legacy_implicit_noexcept if options else False
```
The `getattr()` suggests to me that there are cases where the option might not be initialised, in which case I'd rather see it being initialised.